walk-in-interview/src/main/webapp/account/create-account/business/script.js

```diff
@@ -0,0 +1,221 @@
+
+  const backButton = document.getElementById('back');
+  backButton.innerText = ACCOUNT_STRINGS['back'];
+
+  const submitButton = document.getElementById('submit');
+  submitButton.innerText = ACCOUNT_STRINGS['submit'];
+}
+
+const backButton = document.getElementById('back');
+backButton.addEventListener('click', (_) => {
+  window.location.href = CREATE_ACCOUNT_HOMEPAGE_PATH;
+});
+
+const submitButton = document.getElementById('submit');
+submitButton.addEventListener('click', async (_) => {
+  // Disables the button to avoid accidental double click
+  document.getElementById('submit').disabled = true;
+
+  const account = document.getElementById('account-input').value.trim();
+
+  if (account === '') {
+    setErrorMessage(/* errorMessageElementId= */'error-message',
+        /* msg= */ STRINGS['empty-email'],
+        /* includesDefault= */false);
+    return;
+  }
+
+  const password = document.getElementById('password-input').value.trim();
+
+  // Resets the error (there might have been an error msg from earlier)
+  setErrorMessage(/* errorMessageElementId= */'error-message',
+      /* msg= */ '',
+      /* includesDefault= */false);
+
+  try {
+    await Auth.createBusinessAccount(account, password);
+    await createEmptyPlaceholderAccountObject();
+  } catch (error) {
+    showErrorMessageFromError(error);
+  }
+  
+  // Enables the button regardless of success or failure
+  document.getElementById('submit').disabled = false;
+});
+
+/**
+ * Displays the error messages according to different situations.
+ *
+ * @param {Error} error Error from sign in.
+ */
+function showErrorMessageFromError(error) {
+  switch (error.code) {
+    case EMAIL_IN_USE_ERROR_CODE:
+      setErrorMessage(/* errorMessageElementId= */'error-message',
+          /* msg= */ STRINGS['user-in-use-error'],
+          /* includesDefault= */false);
+      break;
+    case INVALID_EMAIL_ERROR_CODE:
+      setErrorMessage(/* errorMessageElementId= */'error-message',
+          /* msg= */ STRINGS['invalid-email-error'],
+          /* includesDefault= */false);
+      break;
+    case OPERATION_NOT_ALLOWED_ERROR_CODE:
+      setErrorMessage(/* errorMessageElementId= */'error-message',
+          /* msg= */ STRINGS['operation-not-allowed-error'],
+          /* includesDefault= */false);
+      break;
+    case WEAK_PASSWORD_ERROR_CODE:
+      setErrorMessage(/* errorMessageElementId= */'error-message',
+          /* msg= */ STRINGS['weak-password-error'].replace('{MIN_PASSWORD_LENGTH}', MIN_PASSWORD_LENGTH),
+          /* includesDefault= */false);
+      break;
+    default:
+      setErrorMessage(/* errorMessageElementId= */'error-message',
+          /* msg= */ ACCOUNT_STRINGS['error-message'],
+          /* includesDefault= */false);
+      console.error(error.message);
+      break;
+  }
+}
+
+/**
+ * Creates an account object with email as name for the user
+ */
+async function createEmptyPlaceholderAccountObject() {
```

Comment:
I think we can rename this from create empty placeholder to createPreliminaryBusinessAccount?